libs/auth/src/signatures.ts

```diff
@@ -34,3 +34,22 @@ export function constructPrefixedMessage( // eslint-disable-line vx/gts-jsdoc
   message.append(messagePayload);
   return message;
 }
+
+/**
+ * Deconstructs a prefixed message into its components
+ */
+export function deconstructPrefixedMessage(prefixedMessage: string): {
+  messageType: string;
+  messagePayload: string;
+} {
+  const parts = prefixedMessage.split(MESSAGE_SEPARATOR);
+
+  if (parts.length < 3 || parts[0] !== MESSAGE_FORMAT_VERSION.toString()) {
+    throw new Error('Invalid prefixed message format');
+  }
+
+  return {
+    messageType: parts[1] as string,
+    messagePayload: parts[2] as string,
```

Comment:
A bit of a gotcha but it's been hard to guarantee that the message separator doesn't show up in the message payload, so I might make a tweak here to return all parts index 2 and onward concatenated back together just in case there are more than 2 instances of the message separator, e.g., `parts.slice(2, parts.length).join(MESSAGE_SEPARATOR)`

A recent PR where this came up: https://github.com/votingworks/livecheck-web/pull/4